vmx/src/commit.rs

```diff
@@ -147,6 +197,46 @@ impl SessionCommits {
             None => Err(SessionError("unknown session commit id".into())),
         }
     }
+
+    pub fn read<P: AsRef<Path>>(path: P) -> Result<Self, Error> {
```

Comment:
naming here, using "read/write" is very generic and overlaps with the Read and Write traits unnecessarily, consider store/restore or something similar (in all cases where read/write is used)